### PR TITLE
utils_test: install dependency packages for the stress tool

### PR DESCRIPTION
**Pull request by balamuruhans**
stress tool might need some dependencies packages to be installed
before installing the tool, this patch enables a param that hold the
list of packages to be installed.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
Travis failure and this PR depends on: https://github.com/avocado-framework/avocado-vt/pull/1663

**Comment by balamuruhans** [comment]
@vivianQizhu @aliang123 request for your review.

File: virttest/utils_test/__init__.py [hunk 1]
```diff
@@ -1880,6 +1881,8 @@ def __init__(self, stress_type, params, download_url="", make_cmds="",
         :param download_type: currently support "git" or "file" download
         :param downloaded_file_path: Already downloaded / available stress
                                      tool path.
+        :param dependency_packages: prior to install stress tool, its list of
+                                    dependency packages that are to be installed.
```
Comment (vivianQizhu) [inline virttest/utils_test/__init__.py]: Would you like to provide an example format here for the param?

Comment (balamuruhans) [reply virttest/utils_test/__init__.py]: sure, I will provide that.

Comment (balamuruhans) [reply virttest/utils_test/__init__.py]: Done.

**Comment by vivianQizhu** [comment]
Code looks good to me, would you provide a test result?

**Comment by balamuruhans** [comment]
sure, I have tested it for https://github.com/autotest/tp-libvirt/pull/1687 with this patch for stress-ng to work,

```
JOB ID     : 603bad43ff81070e611cc70cd48af4a3e4b858a3
JOB LOG    : /home/bala/avocado-fvt-wrapper/results/job-2018-08-14T17.26-603bad4/job.log
 (1/1) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.HostOS.ppc64le.powerkvm-libvirt.virsh.migrate_stress.orderly_migration.run_stress_app.stress_tool_in_vms.stress_ng.cpu_stress.precopy.set_vcpu_1:  PASS (894.79 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 898.33 s
JOB HTML   : /home/bala/avocado-fvt-wrapper/results/job-2018-08-14T17.26-603bad4/results.html
```

**Log:**
```
2018-08-14 17:29:05,229 virt_vm          L1125 DEBUG| Attempting to log into 'vms3' (timeout 240s)
2018-08-14 17:29:05,229 virt_vm          L0841 DEBUG| Found/Verified IP 192.168.122.215 for VM vms3 NIC 0
2018-08-14 17:29:06,643 virt_vm          L1125 DEBUG| Attempting to log into 'vms3' (timeout 240s)
2018-08-14 17:29:06,645 virt_vm          L0841 DEBUG| Found/Verified IP 192.168.122.215 for VM vms3 NIC 0
2018-08-14 17:29:08,055 __init__         L2004 INFO | Download stress tool from http://kernel.ubuntu.com/~cking/tarballs/stress-ng/stress-ng-0.09.31.tar.xz
2018-08-14 17:29:08,056 download         L0066 INFO | Fetching http://kernel.ubuntu.com/~cking/tarballs/stress-ng/stress-ng-0.09.31.tar.xz -> /var/tmp/avocado_bSLLFK/stress-ng-0.09.31.tar.xz
2018-08-14 17:29:08,563 __init__         L2030 INFO | Copy stress tool to work dir of guest
2018-08-14 17:29:08,564 remote           L0503 DEBUG| Trying to SCP with command 'scp -r -v -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PreferredAuthentications=password  -P 22 /var/tmp/avocado_bSLLFK/stress-ng-0.09.31 root@\[192.168.122.215\]:/home', timeout 600s
2018-08-14 17:29:09,253 remote           L0453 DEBUG| Got password prompt, sending '123456'
2018-08-14 17:29:09,849 remote           L0480 DEBUG| SCP process terminated with status 0
2018-08-14 17:29:09,887 remote           L0871 INFO | Copy file from /var/tmp/avocado_bSLLFK/stress-ng-0.09.31 to 192.168.122.215:/home, elapsed time: 1.32319498062
2018-08-14 17:29:09,887 client           L1126 DEBUG| Sending command: which apt-get
2018-08-14 17:29:09,991 client           L1126 DEBUG| Sending command: echo $?
2018-08-14 17:29:10,092 client           L1126 DEBUG| Sending command: which yum
2018-08-14 17:29:10,197 client           L1126 DEBUG| Sending command: echo $?
2018-08-14 17:29:10,298 client           L1126 DEBUG| Sending command: rpm -q @Development Tools
2018-08-14 17:29:10,419 client           L1126 DEBUG| Sending command: echo $?
2018-08-14 17:29:10,520 client           L1126 DEBUG| Sending command: yum install -y @Development Tools
2018-08-14 17:29:15,378 client           L1126 DEBUG| Sending command: echo $?
2018-08-14 17:29:17,382 __init__         L2048 INFO | make and install stress-ng
2018-08-14 17:29:17,382 client           L1126 DEBUG| Sending command: cd /home/stress-ng-0.09.31/;make && make install
2018-08-14 17:30:13,332 client           L1126 DEBUG| Sending command: echo $?
2018-08-14 17:30:13,434 client           L1162 DEBUG| Sending command (safe): cd /home/stress-ng-0.09.31/
2018-08-14 17:30:13,535 __init__         L1934 INFO | Launch stress with command: nohup stress-ng --cpu 4 --quiet --timeout 3600 > /dev/null &
2018-08-14 17:30:13,535 client           L1162 DEBUG| Sending command (safe): nohup stress-ng --cpu 4 --quiet --timeout 3600 > /dev/null &
2018-08-14 17:30:16,137 utils_misc       L0683 DEBUG| wait for stress app to start (2.000498 secs)
2018-08-14 17:30:16,138 client           L1126 DEBUG| Sending command: pidof -s stress-ng
2018-08-14 17:30:16,271 client           L1126 DEBUG| Sending command: echo $?

```

**Pull request merged**